app/data/tests/test_dedupe_task.py

```diff
@@ -0,0 +1,96 @@
+from datetime import datetime, timedelta
+
+import pytz
+
+from ashlar.models import RecordSchema, RecordType, Record
+from data.models import DedupeJob, RecordDuplicate
+from data.tasks.deduplicate_records import find_duplicate_records
+
+from django.test import TestCase
+from django.db.models import Q
+
+
+class DedupeTaskTestCase(TestCase):
+    def setUp(self):
+        super(DedupeTaskTestCase, self).setUp()
+        self.now = datetime.now(pytz.timezone('Asia/Manila'))
+        self.then = self.now - timedelta(days=10)
+        self.beforeThen = self.then - timedelta(days=1)
+        self.afterThen = self.then + timedelta(days=1)
+        self.beforeNow = self.now - timedelta(days=1)
+        self.afterNow = self.now + timedelta(days=1)
+
+        self.tod = self.now.hour
+        self.dow = self.now.isoweekday() + 1  # 1 added here to handle differences in indexing
+
+        self.record_type = RecordType.objects.create(label='foo', plural_label='foos')
+        self.schema = RecordSchema.objects.create(
+            schema={"type": "object"},
+            version=1,
+            record_type=self.record_type
+        )
+        # 3 identical records to test dedupe
+        self.record1 = Record.objects.create(
+            occurred_from=self.now,
+            occurred_to=self.now,
+            geom='POINT (0 0)',
+            location_text='Equator1',
+            schema=self.schema
+        )
+        self.record2 = Record.objects.create(
+            occurred_from=self.now,
+            occurred_to=self.now,
+            geom='POINT (0 0.00001)',
+            location_text='Equator2',
+            schema=self.schema
+        )
+        self.record3 = Record.objects.create(
+            occurred_from=self.now,
+            occurred_to=self.now,
+            geom='POINT (0 0.0002)',
+            location_text='Equator3',
+            schema=self.schema
+        )
+        # and one that shouldn't be considered a duplicate
+        self.record4 = Record.objects.create(
+            occurred_from=self.now,
+            occurred_to=self.now,
+            geom='POINT (0 5)',
+            location_text='somewhere else1',
+            schema=self.schema
+        )
+
+    def test_find_duplicate_records(self):
```

Comment:
As a general note, I think it would be good to add some tests that directly target your helper functions rather than just the top-level celery job. This guarantees that they're exercised even if the synthetic situation in the test wouldn't trigger all the code paths. It's also a really good gut-check on modularity; if it's easy to call the helper functions from tests then it's a good indication that they're fairly composable.